backend/server/routers/followups.py

```diff
@@ -0,0 +1,90 @@
+from typing import Dict
+from fastapi import APIRouter
+from server.database import coursesCOL
+import json 
+
+router = APIRouter(
+    prefix="/followups",
+    tags=["followups"],
+)
+
+def get_next_term(term: str) -> str:
+    # 'S' -> 'T1'
+    # 'T1' -> 'T2'
+    # 'T2' -> 'T3'
+    # 'T3' -> 'S'
+    if term[0] == 'T':
+        if int(term[-1]) < 3:
+            return 'T' + str(int(term[-1]) + 1)
+        return 'S'
+    return 'T1'
```

Comment:
I think the multiple returns here add more branches in the logic that is required;

Since we want the next element of an ordered set, we can hardcode the term strings as
`["S", "T1", "T2", "T3"]`

Then we can just grab the index of the str given and return the next element (modded by the list to allow cycling)